tests/subsys/net/lib/nrf_cloud/fota_common/src/main.c

```diff
@@ -0,0 +1,716 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <net/nrf_cloud.h>
+#include "fakes.h"
+
+/* This function runs before each test */
+static void run_before(void *fixture)
+{
+	ARG_UNUSED(fixture);
+}
+
+/* This function runs after each completed test */
+static void run_after(void *fixture)
+{
+	ARG_UNUSED(fixture);
+}
+
+ZTEST_SUITE(nrf_cloud_fota_common_test, NULL, NULL, run_before, run_after, NULL);
+
+ZTEST(nrf_cloud_fota_common_test, test_nrf_cloud_fota_is_type_modem)
+{
+	/* Test range beyond enum values */
+	for (enum nrf_cloud_fota_type fota_type = NRF_CLOUD_FOTA_TYPE__FIRST - 1;
+	     fota_type <= NRF_CLOUD_FOTA_TYPE__INVALID + 1;
+	     ++fota_type) {
+
+		if ((fota_type == NRF_CLOUD_FOTA_MODEM_FULL) ||
+		    (fota_type == NRF_CLOUD_FOTA_MODEM_DELTA)) {
+			zassert_true(nrf_cloud_fota_is_type_modem(fota_type),
+				     "Modem FOTA type not detected");
+		} else {
+			zassert_false(nrf_cloud_fota_is_type_modem(fota_type),
+				      "Non-modem FOTA type detected as modem type");
+		}
+	}
+}
+
+#define APP_FOTA_ASSERT_STR "APP FOTA enabled should correspond to the value of: " \
+			    "CONFIG_BOOTLOADER_MCUBOOT"
+
+#define BOOT_FOTA_ASSERT_STR "BOOT FOTA enabled should correspond to the value of: " \
+			     "CONFIG_BOOTLOADER_MCUBOOT + CONFIG_BUILD_S1_VARIANT + " \
+			     "CONFIG_SECURE_BOOT"
+
+#define MDM_DELTA_ASSERT_STR "DELTA MODEM FOTA enabled should correspond to the value of: " \
+			     "CONFIG_NRF_MODEM"
+
+#define FMFU_ASSERT_STR "FULL MODEM FOTA enabled should correspond to the value of: " \
+			"CONFIG_NRF_CLOUD_FOTA_FULL_MODEM_UPDATE + CONFIG_NRF_MODEM"
+
+ZTEST(nrf_cloud_fota_common_test, test_nrf_cloud_fota_is_type_enabled)
+{
+	Z_TEST_SKIP_IFNDEF(CONFIG_FOTA_DOWNLOAD);
```

Comment:
I think you can remove this skip statement, these functions should still give valid results if `CONFIG_FOTA_DOWNLOAD` isn't enabled